homeassistant/components/recorder/__init__.py

```diff
@@ -210,6 +243,19 @@ def purge_ticker(event):
                 self.queue.task_done()
                 continue
 
+            entity_id = event.data.get(ATTR_ENTITY_ID)
+            domain = event.data.get(ATTR_DOMAIN)
+
+            if self.exclude and entity_id in self.exclude_entities or \
+               domain in self.exclude_domains:
+                self.queue.task_done()
+                continue
+
+            if self.include and entity_id not in self.include_entities and \
```

Comment:
Self.include should always be true here, because of voluptuous passing in a dict of empty lists